extensions/rv32im/circuit/src/extension.rs

```diff
@@ -508,6 +521,25 @@ mod phantom {
         }
     }
 
+    impl<F: PrimeField32> PhantomSubExecutor<F> for Rv32HintRandomSubEx {
+        fn phantom_execute(
+            &mut self,
+            _: &MemoryController<F>,
+            streams: &mut Streams<F>,
+            _: PhantomDiscriminant,
+            len: F,
```

Comment:
this seems wrong? when you do `= In len` in the macro, that means here what's passed is register index of `len`. You need to unsafe read from address space 1 to get the actual len.